### PR TITLE
Add support for embedding function

### PR DESCRIPTION
**Pull request by sudeepag**
This PR adds support for an embedding function, which will be helpful for a couple of different use cases
- OpenAI API embedding queries
- Langchain question answering and other use cases
- MiniGPT4 support

The PR is still WIP.

### CODE REVIEW

File: docs/tutorials/runtime/cpp.rst [hunk 1]
```diff
@@ -219,6 +219,7 @@ MLC-compiled models can be integrated into any C++ project using TVM's C/C++ API
         device_id,
       );
       // Step 2. Obtain all available functions in `mlc_llm`
+      PackedFunc embed = mlc_llm->GetFunction("embed");
```
Comment (yzh119) [inline docs/tutorials/runtime/cpp.rst:222]: You don't need to change this file for now, because it's just a demo showing how to use C++ APIs.

Comment (sudeepag) [reply docs/tutorials/runtime/cpp.rst]: Reverted in https://github.com/mlc-ai/mlc-llm/pull/408/commits/ebab0c51c7a682a992f9a00a5e979c1e117d985e.

Comment (yzh119) [reply docs/tutorials/runtime/cpp.rst]: Thank you!

**Comment by Kathryn-cat** [comment]
closed, moved to #482 to incorporate some latest changes in the repo.

**Pull request closed without merging**